src/sdk/model/edge/edge.ts

```diff
@@ -723,6 +729,43 @@ export class Edge extends Entity {
       return new Task(json);
     });
   }
+
+  /**
+   * Get the list of firewall source types
+   * @returns {Promise<Array<EdgeGatewayFirewallObjectType>>}
+   */
+  /* istanbul ignore next: autogenerated */
+  async getFirewallSourceTypes(): Promise<Array<EdgeGatewayFirewallObjectType>> {
+    return Iland.getHttp().get(`/edge-gateways/${this.uuid}/firewall/sources`).then((response) => {
+      const json = response.data.data as Array<EdgeGatewayFirewallObjectTypeJson>;
+      return json.map((it) => new EdgeGatewayFirewallObjectType(it));
+    });
+  }
+
+  /**
+   * Get the edge firewall source objects of the specified type.
+   * If the specified page number is larger than number of actual pages, the response will return as an empty list.
+   * @param {string} type Type. Valid types can be retrieved from the getFirewallSourceTypes() endpoint
+   * @param {number} page The page number. The index of the first page is 1. (Default: 1)
+   * @param {number} pageSize The page size. (Default 25)
+   * @returns {Promise<Array<EdgeGatewayFirewallObject>>}
+   */
+  /* istanbul ignore next: autogenerated */
+  async getFirewallSourceObjects(type: string | EdgeGatewayFirewallObjectType, page: number = 1, pageSize: number = 25):
+      Promise<Array<EdgeGatewayFirewallObject>> {
```

Comment:
your response does not include the `total` parameter, which is going to be important for some clients